File: pset6/readability/readability.py
from cs50 import get_string


# compute sentences
def compute_sentences(sentence):
    # compute number of sentences
    computed_sentences = sentence.count(".") + sentence.count("!") + sentence.count("?")

    # return sentences
    return computed_sentences


# compute words
def compute_words(sentence):
    # initialize needed variables
    computedwords = 0

    # loop through chars in sentences, looking for spaces
    for i in range(len(sentence)):
        if sentence[i] == " ":
            computedwords += 1

    # return words
    computedwords += 1
    return computedwords


# compute letters
def compute_letters(sentence):
    # initialize needed variables
    letters = 0

    # loop through chars in sentences, counting the number of alphabetical chars
    for i in range(len(sentence)):
        if sentence[i].isalpha():
            letters += 1

    # return letters
    return letters


def main():
    # get text from user
    sentence = get_string("Text: ")

    # call functions to get values needed for later
    numSentences = compute_sentences(sentence)
    numWords = compute_words(sentence)
    numLetters = compute_letters(sentence)

    # caculate grade using the Coleman-Liau formula
    L = float(numLetters / numWords * 100)
    S = float(numSentences / numWords * 100)

    grade = round(0.0588 * L - 0.296 * S - 15.8)
    
    # print grade
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    elif grade > 1 and grade < 16:
        print(f"Grade {grade}")


# run main
if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
if #include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>


int compute_sentences(string sentence1)
{
    int computedsentences = 0;
    for (int i = 0, c = strlen(sentence1); i != c; i++)
    {
        if (sentence1[i] == '.')
        {
            computedsentences += 1;
        }

        if (sentence1[i] == '!')
        {
            computedsentences += 1;
        }

        if (sentence1[i] == '?')
        {
            computedsentences += 1;
        }
    }

    return computedsentences;

}

int compute_words(string sentence2)
{
    int computedwords = 0;
    for (int i = 0, c = strlen(sentence2); i != c; i++)
    {
        if (sentence2[i] == ' ')
        {
            computedwords++;
        }
    }

    return computedwords + 1; //+1 because number of words is always number of spaces + 1
}

int compute_letters(string sentence3)
{
    int letters = 0;
    for (int i = 0, c = strlen(sentence3); i != c; i++)
    {
        if (isalpha(sentence3[i]))
        {
            letters += 1;
        }
    }
    return letters;
}


int main(void)
{
    string sentence = get_string("Text: ");

    int numSentences = compute_sentences(sentence); // number of sentences
    int numWords = compute_words(sentence); // number of words
    int numLetters = compute_letters(sentence); // number of letters



    //float lettersaverage = (float) letters / ((float)words * 100.00);
    //float sentencesaverage = (float) sentences / ((float)words * 100.00);

    float L = numLetters / (float)numWords * 100; // Average number of letters per 100 words
    float S = numSentences / (float)numWords * 100; // Average number of sentences per 100 words


    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
    
}
